libs/ledger-live-common/src/families/cosmos/crypto/crypto.ts

```diff
@@ -0,0 +1,305 @@
+      // This correction adds the fees to the rate computation
+      const yearlyFeeRate =
+        (rewardsState.averageDailyFees * 365.24) / rewardsState.totalSupply;
+      return (
+        inexactBlockTimeCorrection *
+        (yearlyInflation + yearlyFeeRate) *
+        (1 / rewardsState.actualBondedRatio) *
+        (1 - rewardsState.communityPoolCommission) *
+        (1 - validatorCommission)
+      );
+    } else if (this.currencyId === "osmo") {
+      return 0.15; // todo fix this obviously
+    } else if (this.currencyId === "juno") {
+      return 0.15; // todo fix this obviously
+    } else {
+      throw new Error(`${this.currencyId} is not supported`);
+    }
+  }
+
+  getRewardsState(): any {
+    if (this.currencyId === "cosmos") {
+      return this.getStargateRewardsState();
+    } else if (this.currencyId === "osmo") {
+      return this.getOsmosisRewardsState();
+    } else if (this.currencyId === "juno") {
+      return this.getOsmosisRewardsState();
+    } else {
+      throw new Error(`${this.currencyId} is not supported`);
+    }
+  }
+
+  private computeAvgYearlyInflation = (rewardsState: CosmosRewardsState) => {
+    // Return invalid rewardsState if
+    // rewardsState.currentValueInflation is not between inflationMinRate and inflationMaxRate
+    const inflationSlope =
+      (1 - rewardsState.actualBondedRatio / rewardsState.targetBondedRatio) *
+      rewardsState.inflationRateChange;
+    const unrestrictedEndOfYearInflation =
+      rewardsState.currentValueInflation * (1 + inflationSlope);
+
+    if (
+      unrestrictedEndOfYearInflation <= rewardsState.inflationMaxRate &&
+      unrestrictedEndOfYearInflation >= rewardsState.inflationMinRate
+    ) {
+      return (
+        (rewardsState.currentValueInflation + unrestrictedEndOfYearInflation) /
+        2
+      );
+    }
+
+    if (unrestrictedEndOfYearInflation > rewardsState.inflationMaxRate) {
+      const diffToMax =
+        rewardsState.inflationMaxRate - rewardsState.currentValueInflation;
+      const maxPoint = diffToMax / inflationSlope;
+      const averageInflation =
+        (1 - maxPoint / 2) * rewardsState.inflationMaxRate +
+        (maxPoint / 2) * rewardsState.currentValueInflation;
+      return averageInflation;
+    }
+
+    if (unrestrictedEndOfYearInflation < rewardsState.inflationMinRate) {
+      const diffToMin =
+        rewardsState.currentValueInflation - rewardsState.inflationMinRate;
+      const minPoint = diffToMin / inflationSlope;
+      const averageInflation =
+        (1 - minPoint / 2) * rewardsState.inflationMinRate +
+        (minPoint / 2) * rewardsState.currentValueInflation;
+      return averageInflation;
+    }
+
+    throw new Error("Unreachable code");
+  };
+
+  private getStargateRewardsState = makeLRUCache(
+    async () => {
+      // All obtained values are strings ; so sometimes we will need to parse them as numbers
+      const inflationUrl = `${this.lcd}/cosmos/mint/${this.version}/inflation`;
```

Comment:
It's seems that inflation rate is not working  and depending on coin so : as we can take a look at what keplr do : 
https://github.com/chainapsis/keplr-wallet/blob/07673165267d0a29d9ab12303299361bdc2e3338/packages/stores/src/query/cosmos/supply/inflation.ts

Reply:
Fixed